src/mod/endpoints/mod_sofia/sofia_reg.c

```diff
@@ -155,6 +155,7 @@ void sofia_reg_fire_custom_gateway_state_event(sofia_gateway_t *gateway, int sta
 	if (switch_event_create_subclass(&s_event, SWITCH_EVENT_CUSTOM, MY_EVENT_GATEWAY_STATE) == SWITCH_STATUS_SUCCESS) {
 		switch_event_add_header_string(s_event, SWITCH_STACK_BOTTOM, "Gateway", gateway->name);
 		switch_event_add_header_string(s_event, SWITCH_STACK_BOTTOM, "State", sofia_state_string(gateway->state));
+		switch_event_add_header(s_event, SWITCH_STACK_BOTTOM, "Failure-Status", "%d", gateway->failure_status);
```

Comment:
The status parameter in `sofia_reg_fire_custom_gateway_state_event` is also the SIP response value. Why do we need to add a new field `Failure-Status` where `status` is also been provided in the event?

Reply:
the `status` provided in the event is an `enum` with the following values

`typedef enum {
	SOFIA_GATEWAY_DOWN,
	SOFIA_GATEWAY_UP,

	SOFIA_GATEWAY_INVALID
} sofia_gateway_status_t;`

Nothing to do with the SIP response value